R/ScenarioConfiguration.R

```diff
@@ -0,0 +1,181 @@
+#' @title ScenarioConfiguration
+#' @docType class
+#' @description An object storing configuration of a specific scenario
+#' @format NULL
+ScenarioConfiguration <- R6::R6Class(
+  "ScenarioConfiguration",
+  inherit = ospsuite.utils::Printable,
+  cloneable = TRUE,
+  active = list(
+    #' @field setTestParameters Boolean representing whether parameters defined in \code{TestParameters} are to be applied
+    #' to the simulation
+    setTestParameters = function(value) {
+      if (missing(value)) {
+        private$.setTestParameters
+      } else {
+        ospsuite.utils::validateIsLogical(value)
+        private$.setTestParameters <- value
+      }
+    },
+    #' @field simulateSteadyState Boolean representing whether the simulation will be brought to a steady-state first
+    simulateSteadyState = function(value) {
+      if (missing(value)) {
+        private$.simulateSteadyState
+      } else {
+        ospsuite.utils::validateIsLogical(value)
+        private$.simulateSteadyState <- value
+      }
+    },
+    #' @field simulationTime Boolean representing whether the simulation will be brought to a steady-state first
+    simulationTime = function(value) {
+      if (missing(value)) {
+        private$.simulationTime
+      } else {
+        ospsuite.utils::validateIsNumeric(value)
+        if (value < 0) {
+          stop(paste0("simulationTime must be a positive numerical value, but the value is ", value))
```

Comment:
This error is being used multiple times, so I am wondering if it would be better to add a new function for it (e.g. `messages$valueShouldBePositive(parameterName)` or `messages$valueShouldNotBeNegative(parameterName)`).

Reply:
Yes this is good